picard/ui/theme.py

```diff
@@ -159,18 +167,29 @@ def update_palette(self, palette, dark_theme, accent_color):
     class MacTheme(BaseTheme):
         @property
         def is_dark_theme(self):
+            dark_theme = False
             if not AppKit:
-                return False
-
-            appearance = AppKit.NSAppearance.currentAppearance()
-            try:
-                basic_appearance = appearance.bestMatchFromAppearancesWithNames_([
-                    AppKit.NSAppearanceNameAqua,
-                    AppKit.NSAppearanceNameDarkAqua
-                ])
-                return basic_appearance == AppKit.NSAppearanceNameDarkAqua
-            except AttributeError:
-                return False
+                return dark_theme
+            if self._loaded_config_theme is not None and self._loaded_config_theme != "System":
+                dark_theme = self._loaded_config_theme == "Dark"
+                # MacOS uses an NSAppearance object to change the current application appearance
+                if dark_theme:
+                    appearance = AppKit.NSAppearance._darkAquaAppearance()
+                else:
+                    appearance = AppKit.NSAppearance._aquaAppearance()
+                AppKit.NSApplication.sharedApplication().setAppearance_(appearance)
```

Comment:
I have not yet tested on macOS, but the most important thing to check here is to verify whether this works with the packaged app or not. Because as it is currently the dark theme selection does not work for the app package.